strr-api/src/strr_api/services/approval_service.py

```diff
@@ -88,8 +88,9 @@ def check_full_name_exists_in_ownership_groups(cls, ltsa_response: LtsaResponse,
         return False
 
     @classmethod
-    def process_auto_approval(cls, token, registration: Registration):
+    def process_auto_approval(cls, token, application: Application):
         """Process approval logic and produce output JSON to store in the DB and providing to FE"""
+        registration = application.registration
```

Comment:
I have made the changes mentioned in your comment. For now I have remove registration status change and save related code from `process_auto_approval`, if this is the right place to create registrations then a function needs to be implemented to process `RegistrationRequest` payload and return `Registration` instance.  I am not sure if you have already started working on it, otherwise I can do it.